HCore-Database/ElasticSearch/Impl/ElasticSearchClientImpl.cs

```diff
@@ -67,34 +73,51 @@ public void Initialize()
                 uriList.Add(uri);
             });
 
-            IConnectionPool connectionPool = 
-                uriList.Count > 1 ? 
-                    (IConnectionPool) new SniffingConnectionPool(uriList) :
-                    (IConnectionPool) new SingleNodeConnectionPool(uriList[0]);
+            NodePool nodePool =
+                uriList.Count > 1 ?
+                    new SniffingNodePool(uriList) :
+                    new SingleNodePool(uriList[0]);
 
-            ConnectionSettings settings;
+            ElasticsearchClientSettings settings;
 
             if (_useJsonNetSerializer)
             {
-                settings = new ConnectionSettings(connectionPool, sourceSerializer: JsonNetSerializer.Default)
+                settings = new ExtendedElasticsearchClientSettings(
+                    nodePool,
+                    sourceSerializer: (serializer, settings) => new NewtonsoftSourceSerializer(serializer),
+                    requestResponseSerializer: (serializer, settings) =>
+                    {
+                        serializer.TryGetJsonSerializerOptions(out var options);
+
+                        options.Converters.Add(new ConcatenateTokenFilterConverter());
+
+                        return serializer;
+                    })
                     .DisableAutomaticProxyDetection()
-                    .ThrowExceptions()
-                    .EnableApiVersioningHeader();
+                    .ThrowExceptions(alwaysThrow: false);
```

Comment:
I set alwaysThrow to false, because during testing a get a few times an exception with a failure
```JSON
"failures": [
    {
        "reason": "[b29jaGlhbGl8fDY3MDg3]: version conflict, required seqNo [9], primary term [1]. current document has seqNo [19] and primary term [1]",
    }
]
```

In the code we look at the object failures rather handling exceptions

```C#
if (deleteByQueryResponse.Failures != null && deleteByQueryResponse.Failures.Any())
	throw new Exception($"Delete by query failed: {deleteByQueryResponse.Failures.First().Index}");

if (deleteByQueryResponse.VersionConflicts > 0)
	throw new Exception($"Delete by query found {deleteByQueryResponse.VersionConflicts} version conflicts");
```

Do you want me to keep this or handle exceptions?

Reply:
This is existing code in IL, lets discuss it there..